From 78fe778fb24b5490c402c3642358ee1fd9d65021 Mon Sep 17 00:00:00 2001
From: Wei-Chiu Chuang <weichiu@cloudera.com>
Date: Mon, 2 May 2016 06:42:51 -0700
Subject: [PATCH 1483/2748] HADOOP-8818. Use equals instead == in
 MD5MD5CRC32FileChecksum and TFileDumper.
 Contributed by Brandon Li.

git-svn-id: https://svn.apache.org/repos/asf/hadoop/common/trunk@1385374 13f79535-47bb-0310-9956-ffa450edef68
(cherry picked from commit 0c53ed4cd122f87b2b041d670b9b59e27da5439c)
(cherry picked from commit 7b64e5805747f7be8ad1b8cbcefc7a312350a199)
(cherry picked from commit 3bcca662bcb2954ebb9d5197b335b75f098ad889)

Change-Id: I7152cd1a650dc6126d9fdc2717282ed8a009a4c8
---
 .../apache/hadoop/fs/MD5MD5CRC32FileChecksum.java  |    2 +-
 .../apache/hadoop/io/file/tfile/TFileDumper.java   |    2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/fs/MD5MD5CRC32FileChecksum.java b/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/fs/MD5MD5CRC32FileChecksum.java
index 5f1fc01..5918995 100644
--- a/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/fs/MD5MD5CRC32FileChecksum.java
+++ b/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/fs/MD5MD5CRC32FileChecksum.java
@@ -134,7 +134,7 @@ public static MD5MD5CRC32FileChecksum valueOf(Attributes attrs
 
     try {
       // old versions don't support crcType.
-      if (crcType == null || crcType == "") {
+      if (crcType == null || crcType.equals("")) {
         finalCrcType = DataChecksum.Type.CRC32;
       } else {
         finalCrcType = DataChecksum.Type.valueOf(crcType);
diff --git a/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/io/file/tfile/TFileDumper.java b/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/io/file/tfile/TFileDumper.java
index f065e62..829a1c6 100644
--- a/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/io/file/tfile/TFileDumper.java
+++ b/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/io/file/tfile/TFileDumper.java
@@ -125,7 +125,7 @@ static public void dumpInfo(String file, PrintStream out, Configuration conf)
           dataSizeUncompressed += region.getRawSize();
         }
         properties.put("Data Block Bytes", Long.toString(dataSize));
-        if (reader.readerBCF.getDefaultCompressionName() != "none") {
+        if (!reader.readerBCF.getDefaultCompressionName().equals("none")) {
           properties.put("Data Block Uncompressed Bytes", Long
               .toString(dataSizeUncompressed));
           properties.put("Data Block Compression Ratio", String.format(
-- 
1.7.9.5

